test/unit/host_test.rb

```diff
@@ -796,6 +796,20 @@ def teardown
       assert_equal ["#{pc} does not belong to the #{h.environment} environment"], h.errors[:puppetclasses]
     end
 
+    test "when saving a host, do not require a puppet environment" do
+      h = FactoryGirl.create(:host, :environment => environments(:production), :puppetmaster => smart_proxies(:puppetmaster))
```

Comment:
Do `FactoryGirl.build(:environment => nil` and test if `h.valid?` (I prefer host to h but to each its own..), that's just so that the test doesn't hit the database, validations are still performed. Same goes for the test below.